geopandas/tools/overlay.py

```diff
@@ -265,23 +265,24 @@ def overlay(df1, df2, how="intersection", keep_geom_type=None, make_valid=True):
                 "df{} contains mixed geometry types.".format(i + 1)
             )
 
-    box_gdf1 = df1.total_bounds
-    box_gdf2 = df2.total_bounds
-
-    if not (
-        ((box_gdf1[0] <= box_gdf2[2]) and (box_gdf2[0] <= box_gdf1[2]))
-        and ((box_gdf1[1] <= box_gdf2[3]) and (box_gdf2[1] <= box_gdf1[3]))
-    ):
-        return GeoDataFrame(
-            [],
-            columns=list(
-                set(
-                    df1.drop(df1.geometry.name, axis=1).columns.to_list()
-                    + df2.drop(df2.geometry.name, axis=1).columns.to_list()
+    if how == "intersection":
+        box_gdf1 = df1.total_bounds
+        box_gdf2 = df2.total_bounds
+
+        if not (
+            ((box_gdf1[0] <= box_gdf2[2]) and (box_gdf2[0] <= box_gdf1[2]))
+            and ((box_gdf1[1] <= box_gdf2[3]) and (box_gdf2[1] <= box_gdf1[3]))
+        ):
+            return GeoDataFrame(
+                [],
+                columns=list(
+                    set(
+                        df1.drop(df1.geometry.name, axis=1).columns.to_list()
+                        + df2.drop(df2.geometry.name, axis=1).columns.to_list()
+                    )
```

Comment:
This whole line of code seems a bit fishy: 1) I am not sure the `set(..)` is actually needed? (the only reason I can think of that is was added is to remove duplicate columns, but a non-empty overlay also doesn't do that .. (since pd.merge preserves duplicate columns)), and 2) this actually also ignores the fact that we add suffixes in the normal code path with merge if there are name clashes between left / right.